### PR TITLE
gnupg => 2.3.4

### PR DESCRIPTION
**Pull request by saltedcoffii**
- Upgrade gnupg and dependencies. works on x86_64. Needs binaries.
- `gettext` rebuild to fix iconv issue with msgmerge complaining that it can't do a conversion.
- Adds `fig2dev`, rebuilds `gettext`, and updates `imagemagick7` as well.

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/saltedcoffii/chromebrew.git CREW_TESTING_BRANCH=gnupg_2.3.4 CREW_TESTING=1 crew update
```

Updated packages are: `imagemagick7 libgpgerror npth libksba libgcrypt gnupg`

### CODE REVIEW

File: packages/gnupg.rb [hunk 1]
```diff
@@ -29,13 +16,20 @@ class Gnupg < Package
...
     system 'make'
   end
 
   def self.install
     system 'make', "DESTDIR=#{CREW_DEST_DIR}", 'install'
   end
+
+  def self.check
+    system 'make', 'check'
+  end
```
Comment (uberhacker) [inline packages/gnupg.rb:34]: :+1:

File: packages/libksba.rb [hunk 2]
```diff
@@ -3,34 +3,25 @@
...
     system 'make'
   end
 
   def self.install
     system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
   end
+
+  def self.check
+    system 'make', 'check'
+  end
```
Comment (uberhacker) [inline packages/libksba.rb:26]: :+1:

File: packages/npth.rb [hunk 3]
```diff
@@ -3,31 +3,22 @@
...
     system "make"
   end
 
   def self.install
     system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
   end
+
+  def self.check
+    system 'make', 'check'
+  end
```
Comment (uberhacker) [inline packages/npth.rb:23]: :+1:

File: packages/libgpgerror.rb [hunk 4]
```diff
@@ -3,25 +3,12 @@
...
-       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libgpgerror/1.42-1_i686/libgpgerror-1.42-1-chromeos-i686.tpxz',
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libgpgerror/1.42-1_x86_64/libgpgerror-1.42-1-chromeos-x86_64.tpxz'
-  })
-  binary_sha256({
-    aarch64: '46ca55616094f55ad7a071f06ea9b028d58ab8883407155ef472cf6b8bc105a7',
-     armv7l: '46ca55616094f55ad7a071f06ea9b028d58ab8883407155ef472cf6b8bc105a7',
-       i686: 'ef29d332a020b339c83492adf00b2aae6173d2a86b4b11922c48338dd73b9138',
-     x86_64: '87bdb815ac35effdf3a42e8dad56f947dcd51280590db3e64b59fc744908ce2d'
-  })
+  source_sha256 'a9ab83ca7acc442a5bd846a75b920285ff79bdb4e3d34aa382be88ed2c3aebaf'
```
Comment (uberhacker) [inline packages/libgpgerror.rb:11]: Add `def self.check`?

File: packages/libgcrypt.rb [hunk 5]
```diff
@@ -3,24 +3,11 @@
...
-  binary_sha256({
-    aarch64: '1e2b42394624c35f94278ee7f9c70aa544c7c40dedd49b396e3e2a72d2e6e6bc',
-     armv7l: '1e2b42394624c35f94278ee7f9c70aa544c7c40dedd49b396e3e2a72d2e6e6bc',
-       i686: 'c072efaef5595c6f3ef7c57ce9573a8c9d79f2e7fa937425f4bb3a4098e12cd5',
-     x86_64: '3945f3464bd06d41db1be4ad4f59242490bfa897856c9e395f25b5d9f1f1bc75'
-  })
+  source_url 'https://www.gnupg.org/ftp/gcrypt/libgcrypt/libgcrypt-1.9.4.tar.bz2'
+  source_sha256 'ea849c83a72454e3ed4267697e8ca03390aee972ab421e7df69dfe42b65caaf7'
 
   depends_on 'libgpgerror'
```
Comment (uberhacker) [inline packages/libgcrypt.rb:12]: Add `def self.check`?

Comment (saltedcoffii) [reply packages/libgcrypt.rb:12]: 5347e5e

File: packages/libgpgerror.rb (hunk 4 above)
Comment (saltedcoffii) [reply packages/libgpgerror.rb:11]: the very first test `gpg-error-config-test.sh` fails and then the tests stop. Ideas?

Comment (uberhacker) [reply packages/libgpgerror.rb:11]: Does the latest crew update help with the tests?

**Comment by satmandu** [comment]
Is this ready to be built or are we still waiting for `def self.check` fixes?

**Comment by saltedcoffii** [comment]
they're ready

**Comment by satmandu** [comment]
Added package and update to package needed for build to continue. (If a build of this also fails, may need to modify the package further.)

**Comment by satmandu** [comment]
Sigh. Failure.
```
rm -f zh_TW.gmo && /usr/local/bin/msgfmt -c --statistics -o zh_TW.gmo zh_TW.po
1794 translated messages, 273 fuzzy translations, 132 untranslated messages.
/usr/local/bin/msgmerge --previous --update zh_CN.po gnupg2.pot
............................................................................................................................................................................................................................ done.
rm -f zh_CN.gmo && /usr/local/bin/msgfmt -c --statistics -o zh_CN.gmo zh_CN.po
2192 translated messages, 3 fuzzy translations, 4 untranslated messages.
make[3]: Leaving directory '/usr/local/tmp/crew/gnupg.20220104042122.dir/po'
touch stamp-po
make[2]: Leaving directory '/usr/local/tmp/crew/gnupg.20220104042122.dir/po'
Making all in doc
make[2]: Entering directory '/usr/local/tmp/crew/gnupg.20220104042122.dir/doc'
convert `test -f 'gnupg-module-overview.svg' || echo './'`gnupg-module-overview.svg gnupg-module-overview.png
convert: no decode delegate for this image format `PNG' @ error/constitute.c/ReadImage/607.
convert: no images defined `gnupg-module-overview.png' @ error/convert.c/ConvertImageCommand/3304.
make[2]: *** [Makefile:1196: gnupg-module-overview.png] Error 1
make[2]: Leaving directory '/usr/local/tmp/crew/gnupg.20220104042122.dir/doc'
make[1]: *** [Makefile:629: all-recursive] Error 1
make[1]: Leaving directory '/usr/local/tmp/crew/gnupg.20220104042122.dir'
make: *** [Makefile:549: all] Error 2
```

This might also need an imagemagick rebuild...

**Comment by satmandu** [comment]
ok rebuild of imagemagick7 looks like it has the right bits...

```
  CPPFLAGS        =  -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16
  PCFLAGS         =
  DEFS            = -DHAVE_CONFIG_H
  LDFLAGS         = -flto=auto -fno-stack-protector -U_FORTIFY_SOURCE
  LIBS            =  -llcms2 -lfreetype    -lxml2 -lflif -lfontconfig -lfreetype -lXext   -lSM -lICE -lX11  -lXt -lbz2 -lz   -lltdl  -lm   -ljemalloc -lpthread
  CXX             = x86_64-cros-linux-gnu-g++
  CXXFLAGS        = -pipe -flto=auto -fno-stack-protector -U_FORTIFY_SOURCE -pthread
  FEATURES        = DPC HDRI Cipher Modules
  DELEGATES       = bzlib flif fontconfig freetype jbig jng jpeg lcms lzma openexr openjp2 pango png ps rsvg tiff video webp wmf x xml zlib zstd
==============================================================================

make  all-am
```

**Comment by satmandu** [comment]
Also @supechicken unrelated but it looks like `make` passing ```-j `nproc` ``` is broken: 
```
pstree -a | grep make
  |   |   |           |       `-make
  |   |   |           |           `-make all-am
  |   |   |           |       `-sh -c bash -e -c "make"
  |   |   |           |           `-make
  |   |   |           |               `-make all-am
  |       |-grep --color=auto make
```

**Comment by satmandu** [comment]
Turns out imagemagick was being broken by the build process removing `.la` files it needs to work.

https://github.com/skycocker/chromebrew/pull/6551 fixes that build issue.

File: packages/libgpgerror.rb (hunk 4 above)
Comment (satmandu) [reply packages/libgpgerror.rb:11]: Added a self-check option.

**Comment by satmandu** [comment]
@uberhacker This should be good to go.

**Pull request merged**